### PR TITLE
CR013: add KeyList and TypesOfValue structures

### PR DESCRIPTION
**Pull request by haeckerbaer**
Change Request for the addition of the NeTEx keyList and TypeOfValue structures to the Siri Extensions structure. See documentation in the basecamp:
[Siri_CR_013_KeyList_ValueSet.doc](https://3.basecamp.com/3256016/buckets/9672657/uploads/1561466875)

Keep in mind that this commit only adds the abstract part of the ValueSet / TypeOfValue structures. To actually use the TypeOfValues one has to define the non-abstract types, as for example TypeOfProductCategory in NeTEx, which is then based on the TypeOfValueStructure. KeyList, on the other hand, can be used as is. It has to be discussed what TypeOfXXX types will be needed in the future.

### CODE REVIEW

**Comment by Aurige** [comment]
Why did you aslo add the TypesOfValue, it's not necessary here (and that's another topic, I don't think we have CR about it)  ?
Also from that point we need to add the KeyList inside the SIRI extension structure

**Comment by haeckerbaer** [comment]
@Aurige 
- It was communicated to me that the ValueSet structure (and therefore the TypeOfValue structure as well) are also requested.
- the KeyList structure is available within the Extensions structure and it is correclty validated. You can try it out in an example.
- I can remove the ValueSet structure if it is requested, but the masterlist reads "accepted" for my proposal

**Comment by Aurige** [comment]
Ok for the TypeOfValue... that's anyway necessary for the Branding

File: xsd/siri_utility/siri_utility-v1.1.xsd [hunk 1]
```diff
@@ -80,4 +80,215 @@ Rail transport, Roads and road transport
...
+  </xsd:sequence>
+ </xsd:complexType>
+ <!-- ======================================================================= -->
+ <xsd:simpleType name="TypeOfValueIdType">
+  <xsd:annotation>
+   <xsd:documentation>Type for identifier of a TYPE OF VALUE.</xsd:documentation>
+  </xsd:annotation>
+  <xsd:restriction base="ObjectIdType"/>
+ </xsd:simpleType>
+ <xsd:simpleType name="VslueSetIdType">
```
Comment (hennoit) [inline xsd/siri_utility/siri_utility-v1.1.xsd]: change "VslueSetIdType" to "ValueSetIdType"

**Comment by WickieCH** [comment]
I would prefer to add this functionality in the real structure, not in an extension.

**Comment by Aurige** [comment]
@WickieCH Key/values are basically extension, and puting them in the extension structure is also a good way to make it available everywhere (if not we would have to manually add it in every single place where we think it may be of use some day => meaning tens of change in XSD and doc ...)

**Comment by haeckerbaer** [comment]
@hennoit 
Typo was fixed in latest commit.

**Comment by haeckerbaer** [comment]
See latest commits (3c4b7aa etc.) with the revision of TypeOfValue & ValueSet structures. It must be discussed wether TypeOfValue should be abstract, and any actual TypeOfXXX types need to be added later on, as for example TypeOfMonitoringError. Or if TypeOfValue is not abstract, and the class of which they are an instance of is specified within ClassOfValues and/or NameOfClass. The latter is how I modelled it in the latest commits, but different than the NeTEx approach.

**Comment by Aurige** [comment]
Don't we need to add 	`<xsd:include schemaLocation="siri_types-v2.0.xsd"/>`
to be able to validate the file standalone (due to the use of _NaturalLanguageStringStructure_ ) ?

**Comment by haeckerbaer** [comment]
@Aurige 
Was included in latest commit.

@hennoit 
Annotations were updated in latest commit.

**Pull request merged**